apps/frontend/src/app/3_organisms/BorrowLoan/components/AdjustLoanForm/hooks/useGetMaximumBorrowAmount.ts

```diff
@@ -0,0 +1,102 @@
+import { useGetRBTCPrice } from '../../../../../../hooks/zero/useGetRBTCPrice';
+import { decimalic } from '../../../../../../utils/math';
+import { calculatePrepaidInterest } from '../../../../BorrowLoanForm/components/NewLoanForm/NewLoanForm.utils';
+import { useGetMaximumCollateralAmount } from '../../../../BorrowLoanForm/components/NewLoanForm/hooks/useGetMaximumCollateralAmount';
+import { normalizeToken } from '../AdjustLoanForm.utils';
+import { useGetBorrowingAPR } from './useGetBorrowingAPR';
+import { useGetCollateralAssetPrice } from './useGetCollateralAssetPrice';
+
+export const useGetMaximumBorrowAmount = (
+  loan: LoanItem,
+  collateralAmount?: Decimal,
+) => {
+  const borrowToken = useMemo(
+    () => normalizeToken(loan.debtAsset.toLowerCase()),
+    [loan.debtAsset],
+  );
+
+  const collateralToken = useMemo(
+    () => normalizeToken(loan.collateralAsset.toLowerCase()),
+    [loan.collateralAsset],
+  );
+
+  const { weiBalance: borrowAssetBalance } = useMaxAssetBalance(borrowToken);
+
+  const { maximumCollateralAmount } = useGetMaximumCollateralAmount(
+    collateralToken,
+    collateralAmount,
+  );
+
+  const collateral = useMemo(
+    () =>
+      !collateralAmount || collateralAmount.isZero()
+        ? maximumCollateralAmount
+        : collateralAmount,
+    [collateralAmount, maximumCollateralAmount],
+  );
+
+  const debt = useMemo(() => loan.debt, [loan.debt]);
+
+  const { borrowApr } = useGetBorrowingAPR(borrowToken, Decimal.from(debt));
+
+  const { price: rbtcPrice } = useGetRBTCPrice();
+  const { borrowPriceUsd, collateralPriceUsd } = useGetCollateralAssetPrice(
+    borrowToken,
+    collateralToken,
+  );
+
+  const collateralPriceInLoanAsset = useMemo(
+    () =>
+      decimalic(
+        collateralToken === SupportedTokens.rbtc
+          ? rbtcPrice
+          : collateralPriceUsd,
+      ).div(borrowToken === SupportedTokens.rbtc ? rbtcPrice : borrowPriceUsd),
+    [
+      borrowToken,
+      borrowPriceUsd,
+      collateralToken,
+      collateralPriceUsd,
+      rbtcPrice,
+    ],
+  );
+
+  const maximumLoanToCollateralRatio = useMemo(
+    () => Decimal.ONE.div(MINIMUM_COLLATERAL_RATIO_BORROWING_MAINTENANCE),
+    [],
+  );
+
+  const maxBorrow = useMemo(
+    () =>
+      collateral
+        .mul(collateralPriceInLoanAsset)
+        .mul(maximumLoanToCollateralRatio),
+    [collateralPriceInLoanAsset, collateral, maximumLoanToCollateralRatio],
+  );
+
+  const prepaidInterest = useMemo(
+    () => calculatePrepaidInterest(borrowApr, maxBorrow, loan.rolloverDate),
+    [borrowApr, loan.rolloverDate, maxBorrow],
+  );
+
+  const result: Decimal = useMemo(
+    () => maxBorrow.sub(prepaidInterest).sub(debt),
+    [debt, maxBorrow, prepaidInterest],
+  );
+
+  if (result.lte(Decimal.ZERO)) {
+    return Decimal.ZERO;
+  }
+
+  return result.gt(Decimal.fromBigNumberString(borrowAssetBalance))
+    ? Decimal.fromBigNumberString(borrowAssetBalance)
+    : result;
```

Comment:
This does not seems right. Now, if result > my balance, then it only allows to take loan equal to my balance.

<img width="404" alt="Screenshot 2023-09-01 at 08 55 39" src="https://github.com/DistributedCollective/sovryn-dapp/assets/69458664/a8cb5e77-f949-41ac-9589-8c217f2189c6">

My MAX borrow should be something like 6 BTC here

Reply:
Great catch @creed-victor!